scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1084,19 +1084,64 @@
                               :first-ten-jobs (print-str (->> jobs (take 10) (map :job/uuid) (map str)))}))
       (dissoc compute-cluster->jobs :no-acceptable-compute-cluster))))
 
+(defn decrement-scheduling-capacity-counter
+  "Decrement capacity for compute cluster in the counter."
+  [capacity-counter compute-cluster]
+  (update-in capacity-counter [compute-cluster] dec))
+
+(defn scheduling-capacity-constrained-job-distributor
```

Comment:
Can you put a unit test over this code that exercises some of the code paths?

Reply:
done, tested that it ignores clusters with no capacity and dec's the counter.